pkg/clusterconfig/clusterconfig.go

```diff
@@ -156,3 +172,74 @@ func GetGCSConfig() (*Config, error) {
 	cfg := &Config{}
 	return cfg, nil
 }
+
+// GetSwiftConfig reads credentials
+func GetSwiftConfig(listers *regopclient.Listers) (*Config, error) {
+	cfg := &Config{}
+
+	// Look for a user defined secret to get the Swift credentials
+	sec, err := listers.Secrets.Get(imageregistryv1.ImageRegistryPrivateConfigurationUser)
+	if err != nil && errors.IsNotFound(err) {
+		// If no user defined credentials were provided, then try to find them in the secret,
+		// created by cloud-credentials-operator
+		pollErr := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (stop bool, err error) {
+			sec, err = listers.Secrets.Get(cloudCredentialsName)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, err
+			}
+			return true, nil
+		})
+
+		if sec == nil || pollErr != nil {
+			return nil, fmt.Errorf("unable to get cluster minted credentials %q: %v", fmt.Sprintf("%s/%s", imageregistryv1.ImageRegistryOperatorNamespace, cloudCredentialsName), pollErr)
+		}
+
+		if cloudsData, ok := sec.Data["clouds.yaml"]; ok {
```

Comment:
I'd add a comment here that the cloud credential minter is responsible for generating the `clouds.yaml` file and placing it in the cloud creds secret. This will be good to reference if down the line the install team/auth team changes this behavior.

Reply:
done